tests/test_potentials.py

```diff
@@ -453,3 +453,189 @@ def test_bound_and_unbound_execute_match(harmonic_bond_test_system, precision):
         )
         np.testing.assert_array_equal(bound_du_dx, unbound_du_dx)
         np.testing.assert_array_equal(bound_u, unbound_u)
+
+
+def test_execute_batch_sparse_validation(harmonic_bond: BoundPotential[HarmonicBond]):
+    unbound_impl = harmonic_bond.potential.to_gpu(np.float32).unbound_impl
+
+    # Should verify that number of boxes and coords match
+    with pytest.raises(RuntimeError) as e:
+        _ = unbound_impl.execute_batch_sparse(
+            np.zeros((5, 4, 3)),
+            np.zeros((4, 3, 2)),
+            np.zeros((6, 3, 3)),  # inconsistent length
+            np.zeros(3).astype(np.uint32),
```

Comment:
Yes, although this is a `TypeError` raised by the pybind11 wrapper. Leaning toward omitting this case since it seems like the domain of pybind11 and we don't test it elsewhere, but could be convinced otherwise.